### PR TITLE
feat: load attached fonts properly

### PR DESCRIPTION
**Pull request by noaione**
Now attached fonts should load and not falling back to Liberation Sans

**Preview**
image

### CODE REVIEW

File: frontend/src/store/playerElement.ts [hunk 1]
```diff
@@ -115,6 +122,17 @@ class PlayerElementStore {
     }
   };
 
+  private _isSupportedFonts = (mimeType: string): boolean => {
+    // ttf,otf,woff,woff2
```
Comment (ferferga) [inline frontend/src/store/playerElement.ts:126]:
```suggestion
  private _isSupportedFont = (mimeType: string): boolean => {
```

We're checking a single font here
The comment is useless given how concise the function is, imo.

File: frontend/src/store/playerElement.ts [hunk 2]
```diff
@@ -140,6 +158,14 @@ class PlayerElementStore {
         }
       }
 
+      const remote = useRemote();
```
Comment (ferferga) [inline frontend/src/store/playerElement.ts:161]: Move this to the top of the function please

File: frontend/src/store/playerElement.ts [hunk 3]
```diff
@@ -140,6 +158,14 @@ class PlayerElementStore {
...
+
+      let serverAddress = '';
+
+      if (remote.sdk.api) {
+        serverAddress = remote.sdk.api.basePath as string;
+      }
```
Comment (ferferga) [inline frontend/src/store/playerElement.ts:167]:
```suggestion
      const serverAddress = remote.sdk.api?.basePath;
```

File: frontend/src/store/playerElement.ts [hunk 4]
```diff
@@ -154,6 +180,19 @@ class PlayerElementStore {
         (sub) => sub.srcIndex === playbackManager.currentSubtitleStreamIndex
       );
 
+      const attachedFonts =
+        playbackManager.currentMediaSource?.MediaAttachments?.filter((attach) =>
+          this._isSupportedFonts(attach.MimeType || '')
```
Comment (ferferga) [inline frontend/src/store/playerElement.ts:185]:
```suggestion
        playbackManager.currentMediaSource?.MediaAttachments?.filter((a) =>
          a.MimeType ? this._isSupportedFonts(a.MimeType) : false
```

I think this one is less obscure for a filter function

File: frontend/src/store/playerElement.ts [hunk 5]
```diff
@@ -154,6 +180,19 @@ class PlayerElementStore {
...
+        )
+          .map((attach): string | undefined => {
+            if (attach.DeliveryUrl && serverAddress) {
+              return `${serverAddress}${attach.DeliveryUrl}`;
```
Comment (ferferga) [inline frontend/src/store/playerElement.ts:189]:
```suggestion
          .map((a) => {
            if (a.DeliveryUrl && serverAddress) {
              return `${serverAddress}${a.DeliveryUrl}`;
```
To keep it consistent from above.
I also think the type annotations for the return are unnecessary in this case, but ignore that change if locally you get linter complaints.

File: frontend/src/store/playerElement.ts [hunk 6]
```diff
@@ -154,6 +180,19 @@ class PlayerElementStore {
...
+            }
+          })
+          .filter((attach) => attach !== undefined && attach !== null) as
+          | string[]
+          | undefined;
```
Comment (ferferga) [inline frontend/src/store/playerElement.ts:194]:
```suggestion
          .filter((a): a is string => a !== undefined && a !== null) ?? [];
```

Use a type guard so types are correct after filtering

File: frontend/src/store/playerElement.ts [hunk 7]
```diff
@@ -94,13 +94,20 @@ class PlayerElementStore {
...
         prescaleFactor: 0.8,
         onDemandRender: false
       });
     } else if (jassub) {
+      if (Array.isArray(loadedFonts)) {
+        for (const font of loadedFonts) {
+          jassub.addFont(font);
+        }
+      }
+
```
Comment (ferferga) [inline frontend/src/store/playerElement.ts:110]:
```suggestion
    } else if (jassub && Array.isArray(loadedFonts)) {
        for (const font of loadedFonts) {
          jassub.addFont(font);
        }
```

Also, are you sure this is ever going to be needed? I'm on mobile right now, but iirc we created/destroyed jassub every time the currently playing item changes? Or maybe just on unmount?

**Review by ferferga — CHANGES_REQUESTED** [review]
Thank you very much for your contribution! 🙌

Just checked and I didn't see you in our Matrix room, so take this as an invitation to join. It's more easier to get in touch there for planning/dev questions.

This is pretty good overall, but some minor nitpicky stuff to (probably) keep the complexity level below the threshold!

**Comment by noaione** [comment]
Applied suggestion, somehow sonarlint complexity check got worse though. Also joined the matrix channels.

**Review by ferferga — APPROVED** [review]

**Pull request merged**